### PR TITLE
Make closing dialogs callback with false

### LINKED ISSUE #21627: Can not listen to "X" close button on `OC.dialogs.confirmDestructive`
From https://github.com/nextcloud/spreed/pull/3591#pullrequestreview-439053284

> When the dialog to leave or join is shown clicking on the close button in the top right closes the dialog without selecting any option, which leaves the UI in a weird state. Unfortunately [the close button is shown by default](https://github.com/nextcloud/server/blob/b1f74b0056b76402fbd618faff817432dec77ee3/core/src/jquery/ocdialog.js#L28) in the dialogs and [there is no parameter to disable it in `OC.dialogs`](https://github.com/nextcloud/server/blob/be0ecbc8c7f855cb0c9822233e38a9b459b80b58/core/src/OC/dialogs.js#L626-L630), so this needs to be fixed in the server.

The `OC.dialogs.confirmDestructive` dialog has a non-removable close button which can not be listened to.
In Talk we a workaround we added:
```js
	const interval = setInterval(function() {
		// eslint-disable-next-line no-undef
		if ($('.oc-dialog-dim').length === 0) {
			clearInterval(interval)
			EventBus.$emit('duplicateSessionDetected')
			window.location = generateUrl('/apps/spreed')
		}
	}, 3000)
```

This is as bad as it can get. Please help us to get rid of it and somehow trigger an event/callback or something so we can listen to the `X` properly (or even remove the `X`).

cc @nextcloud/javascript

### PR DESCRIPTION
**Pull request by gary-kim**
Closes #21627 

@nickvergessen Would this work for you?

Signed-off-by: Gary Kim <gary@garykim.dev>

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]
Looks good, will test run tomorrow

**Review by ChristophWurst — APPROVED** [review]

File: core/src/OC/dialogs.js [hunk 1]
```diff
@@ -625,6 +625,7 @@ const Dialogs = {
 
 			$(dialogId).ocdialog({
 				closeOnEscape: true,
+				closeCallback: () => { callback(false) },
```
Comment (danxuliu) [inline core/src/OC/dialogs.js:628]: This should be guarded against an undefined callback (which is unlikely to happen, though, but for consistency with the [handling of button clicks above](https://github.com/nextcloud/server/blob/be0ecbc8c7f855cb0c9822233e38a9b459b80b58/core/src/OC/dialogs.js#L566-L568)).

**Review by danxuliu — COMMENTED** [review]
Tested and works :+1:

Some message dialogs like [`alert` or `info`](https://github.com/nextcloud/server/blob/be0ecbc8c7f855cb0c9822233e38a9b459b80b58/core/src/OC/dialogs.js#L48-L74) only have an OK button. I think that calling the callback with `false` is fine in those cases too, as in those cases I do not think that the value passed to the callback is relevant, only that the callback is called, and thus I guess that nobody checked whether the callback was `true` or not.

File: core/src/OC/dialogs.js (hunk 1 above)
Comment (gary-kim) [reply core/src/OC/dialogs.js:628]: That's a good point.

**Review by danxuliu — APPROVED** [review]

**Pull request merged**